src/stan_math_backend/Transform_Mir.ml

```diff
@@ -349,10 +366,13 @@ let gen_write_unconstrained (decl_id, sizedtype) =
     {Expr.Typed.Meta.empty with type_= SizedType.to_unsized sizedtype}
   in
   let expr = Expr.Fixed.{meta; pattern= Var decl_id} in
-  let writefn var =
-    Stmt.Helpers.for_scalar_inv
-      (SizedType.inner_type sizedtype)
-      bodyfn var Location_span.empty
+  let writefn st var =
```

Comment:
Tuple constraints/constraint refactoring.

This was written before #856 and #872; it should be significantly easier now, but we'll still need to generate some loops for the Stmt.Helpers.for_X functions.